### PR TITLE
Translation units should compile with -std=c++11 by default

### LINKED ISSUE #124: Translation Units in `src` are compiled with `-std=c++14` by default
## Description

Generated `compile_commands` reveal that translation units in `src` are compiled with `-std=c++14` when user does not explicitly set `CMAKE_CXX_STANDARD`. This is surprising because we set `CMAKE_CXX_STANDARD` to 11 if this macro is not already defined by user. See https://github.com/trytouca/touca-cpp/blob/main/CMakeLists.txt#L34-L37.

## Environment

Behavior appears irrelevant to the choice of OS, compiler, and CMake generator. `-std=c++14` confirmed both on macOS and Linux for the latest code in `main`.

## Steps To Reproduce

Run `./build.sh --all` to build components. Check `local/build/compile_commands.json` and grep for `-std=c++14`

## Expected Behavior

We think the correct flag should be `-std=c++11`.

### PR DESCRIPTION
**Pull request by ghorbanzade**
Removes compile feature cxx_aggregate_default_initializers that fixes #124.

### CODE REVIEW

File: include/touca/devkit/platform.hpp [hunk 1]
```diff
@@ -11,6 +11,8 @@
 namespace touca {
 
 struct Response {
+  Response(const int status, const std::string body)
```
Comment (mapron) [inline include/touca/devkit/platform.hpp:14]: that's pretty unoptimal pattern.
It's good to have const std::string member, but you should have 
const std::string& body
or even better
std::string body + std::move in field init.

File: src/devkit/comparison.cpp [hunk 2]
```diff
@@ -265,19 +265,23 @@ TypeComparison compare(const data_point& src, const data_point& dst) {
 }
 
 std::string Cellar::stringify(const detail::internal_type type) const {
-  const std::unordered_map<detail::internal_type, std::string> store = {
```
Comment (mapron) [inline src/devkit/comparison.cpp:268]: Wow, that was pretty inefficient version here, actually.

Comment (mapron) [reply src/devkit/comparison.cpp:268]: it craetes local map on stack, fills it with values, then get only one value (I think even plain std::map would be much better case in that scenario, if we can't use static map).

Again, static (global lifetime, I'd say) objects must be avoided unless its constexpr (then we fine).
But I don't think constexpr containers are easy (like. frozen containers starts only from C++14, and I have a problems with MSVC with them until I got to C++17).

well, that's offtopic. switch for the good.

File: include/touca/devkit/platform.hpp (hunk 1 above)
Comment (ghorbanzade) [reply include/touca/devkit/platform.hpp:14]: Yeah, thanks for catching this. I didn't notice that type of `body` is missing is a reference.

**Pull request merged**